src/scripts/main.js

```diff
@@ -1 +1,75 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', (e) => {
+    if (e.button === 0) {
+      resolve('First promise was resolved');
+    }
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve) => {
+  document.addEventListener('click', (e) => {
+    if (e.button === 0) {
+      resolve('Second promise was resolved');
+    }
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    if (e.button === 2) {
+      e.preventDefault();
+      resolve('Second promise was resolved');
+    }
+  });
+});
+
+const thirdPromise = new Promise((resolve) => {
+  let leftClicked = false;
+  let rightClicked = false;
+
+  const checkCompletion = () => {
+    if (leftClicked && rightClicked) {
+      resolve('Third promise was resolved');
+    }
+  };
+
+  document.addEventListener('click', (e) => {
+    if (e.button === 0) {
+      leftClicked = true;
+      checkCompletion();
+    }
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    if (e.button === 2) {
+      rightClicked = true;
+      checkCompletion();
+    }
+  });
+});
```

Comment:
The third promise resolves when both a left click and a right-click occur. However, if the user performs these actions in quick succession, the promise might resolve multiple times. Consider adding a mechanism to ensure it resolves only once.